include/flox/core/util.hh

```diff
@@ -184,13 +247,56 @@ parseFlakeRef( const std::string & flakeRef )
 /* -------------------------------------------------------------------------- */
 
 /**
- * Parse a JSON object from an inline string or a path to a JSON file.
+ * @brief Parse a JSON object from an inline string or a path to a JSON file.
  * @param jsonOrPath A JSON string or a path to a JSON file.
  * @return A parsed JSON object.
  */
 nlohmann::json parseOrReadJSONObject( const std::string & jsonOrPath );
 
 
+/* -------------------------------------------------------------------------- */
+
+/** @brief Convert a TOML string to JSON. */
+nlohmann::json tomlToJSON( std::string_view toml );
+
+
+/* -------------------------------------------------------------------------- */
+
+/** @brief Convert a YAML string to JSON. */
+nlohmann::json yamlToJSON( std::string_view yaml );
+
+
+/* -------------------------------------------------------------------------- */
+
+/**
+ * @brief Split an attribute path string.
+ *
+ * Handles quoted strings and escapes.
+ */
+std::vector<std::string> splitAttrPath( std::string_view path );
+
+
+/* -------------------------------------------------------------------------- */
+
+/**
+ * @brief Is the string @str a positive natural number?
```

Comment:
Biggest nitpick of all nitpicks: Apparently there's disagreement over whether natural numbers include 0. Mathematicians say no, but apparently there's an ISO standard that says yes. I lean towards the math side and would call this a "positive integer".